index.js

```diff
@@ -45,6 +45,42 @@ app.get('/', function(req, res) {
   res.render('index');
 });
 
+function sendMachines(req, res, token, machines) {
+  var content = {
+    token: token,
+    machines: machines
+  };
+  res.send(content);
+}
+
+app.get('/devices/:token', function(req, res) {
+  return showMachines(req, res, req.params.token);
+});
+
+// get machines for the token and send them along with the token
+function showMachines(req, res, token) {
+  if (!token) {
+    throw('No token provided');
+  }
+  var machines = {};
+  var machineId;
+  function machinePromise(machineId) {
+    return new Promise(function(resolve, reject) {
+      redis.hgetall(machineId)
+        .then(function(machine) {
+          machines[machineId] = machine;
+          resolve();
+        });
+    });
+  }
+  redis.smembers(token)
+    .then(function(ids) {
+      var promises = ids.map(machinePromise);
+      Promise.all(promises)
+        .then(() => sendMachines(req, res, token, machines));
```

Comment:
Can you inline `sendMachines` here and rename the function from `showMachines` to `sendMachines`?